### ISSUE TITLE
Unsaved changes dialog not displayed when using back button

### ISSUE THREAD

**Issue by wonder-sk** [MerginMaps/mobile#3525]
When I open edit form for a feature, do some edits and then press android's system back button, it will show the "Unsaved changes" dialog as expected - but if I use the top-left back button in the form, no warning gets shown and the changes are discarded...

**Comment by jozef-budac** [comment]
This doesn't work well on 2024.3.1 RC at least in two scenarios

1) when the feature form is empty, the android and top left back button should close the window without any notification or confirmation, but

- top left button shows "Unsaved changes" window - bug
- The Android button does nothing - bug

2)  when I fill out the form and hit the top left back button, the "Unsaved changes" window appears. When I click the 'X' button on it to close the modal windows, from not the back Android button does nothing - bug

For the second point, here is the video:
https://github.com/user-attachments/assets/c3457b63-9b83-416f-8e69-1a7678a3897a

**Comment by tomasMizera** [comment]
> This doesn't work well on 2024.3.1 RC at least in two scenarios
> 
>     1. when the feature form is empty, the android and top left back button should close the window without any notification or confirmation, but
> 
> 
>     * top left button shows "Unsaved changes" window - bug
> 
>     * The Android button does nothing - bug
> 
> 
>     2. when I fill out the form and hit the top left back button, the "Unsaved changes" window appears. When I click the 'X' button on it to close the modal windows, from not the back Android button does nothing - bug
> 
> 
> For the second point, here is the video: https://github.com/user-attachments/assets/c3457b63-9b83-416f-8e69-1a7678a3897a

Hi @jozef-budac thanks for that, we will have a look at the android button issue. The "unsaved changes" window should be there when creating a new form all the time (even if you did not change anything) as going back will discard the entire captured geometry.

**Comment by ValentinBuira** [comment]
Hi @jozef-budac  thanks for the catch 

For 1, the android button bug doing nothing  when adding the feature form is empty is already present in last release 
And for 2, now the issue also appear after closing "Unsaved changes" dialog

Both got the same root issue; the *back event is handled only when the form is in focus*, so if you tap on a field again it works again

I'm working on a fix

**Comment by tomasMizera** [comment]
> @tomasMizera , the "Unsaved changes" window appears when hitting the back button even for nongeo feature with an empty. form. Are we OK with that?
> Screen_Recording_20240807_082826.mp4

Yeah, I think that is fine for now, thanks for double-checking @jozef-budac

**Comment by saberraz** [comment]
This is again an issue see, this project:
https://app.merginmaps.com/projects/lutraconsulting/problem_with_Hlink_and_back/tree

**Comment by saberraz** [comment]
To replicate, open the project above, click on a polygon and then open form, once you press Back, it shows you the message.
